peps/pep-0758.rst

```diff
@@ -0,0 +1,432 @@
+`dependency confusion attacks
+<https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610>`__ of
+which Python *can* be particularly vulnerable, due to the algorithm that ``pip
+install`` uses for resolving package dependencies and preferred versions. The
+``uv`` tool addresses this by supporting an additional `index strategy
+<https://docs.astral.sh/uv/reference/settings/#index-strategy>`__ option,
+whereby users can select between, e.g. a ``pip``-compatible strategy, and a
+more limited strategy that prevents such dependency confusion attacks.
+
+:pep:`708` provides additional background about dependency confusion attacks,
+and takes a different approach to preventing them. At its core, PEP 708 allows
+repository owners to indicate that projects track across different
+repositories, which allows installers to determine how to treat the global
+package namespace when combined across multiple repositories. PEP 708 has been
+provisionally accepted, pending several required conditions as outlined in PEP
+708, some of which may have an indeterminate future. As PEP 708 itself says,
+this won't by itself solve dependency confusion attacks, but is one way to
+provide enough information to installers to help minimize these attacks.
+
+While there can still be valid use cases for standing up a totally independent
+package index (such as providing richer platform support for GPUs until a
+fully formed `variant proposal
+<https://discuss.python.org/t/selecting-variant-wheels-according-to-a-semi-static-specification/53446>`__
+is accepted), this PEP takes a different, simpler approach and doesn't replace
+any of the existing, proposed, or approved package index cooperation
+specifications.
+
+This PEP also preserves the core purpose of PyPI, and allows it to
+remain the traditional, canonical, centralized index of all Python
+packages.
+
+It's not just about indexes
+===========================
+
+The problem *this* PEP solves is not just reduction or elimination of
+dependency confusion attacks. There is another class of problem that a safe
+external wheel hosting solution can mitigate: artifact size limits.
+
+PyPI imposes a `default artifact size limit
+<https://pypi.org/help/#file-size-limit>`__ of 100MiB and a default overall
+`project size limit <https://pypi.org/help/#project-size-limit>`__ of
+10GiB. Most packages and artifacts can easily fit in these limits, even for
+packages containing binary extension modules for a variety of platforms. A
+small, but important class of packages routinely exceed these limits,
+requiring them to submit PyPI support tickets `requesting higher limits
+<https://github.com/pypi/support/issues?q=is%3Aissue+is%3Aclosed+file+limit+request>`__.
+It's not necessarily difficult to get resolution on such exceptions, but it is
+a special process that can take some time to resolve, and the criteria for
+granting such exceptions isn't well documented.
```

Comment:
May be good to point out that this process makes more work for the already busy PyPi admins, ideally, they wouldn't have to interact with things like this!

Reply:
```suggestion
granting such exceptions isn't well documented. Furthermore, this process creates more work for the PyPi admins who are already overtaxed with other PyPi-related requests.
```

Maybe something like this?

Reply:
Possibly, although I'm still technically proposing some admin interaction, i.e. to approve org account external hosting.  So it reduces but doesn't fully eliminate admin load.  OTOH, we could auto-approve orgs once they add a support URI.

Reply:
Ah I think I meant to express that we're reducing the admin load, which I think you actually cover below. I'll resolve this :)